deals/rpc/rpc.go

```diff
@@ -192,3 +192,81 @@ func (s *RPC) Retrieve(req *RetrieveRequest, srv RPCService_RetrieveServer) erro
 		}
 	}
 }
+
+// FinalDealRecords calls deals.FinalDealRecords.
+func (s *RPC) FinalDealRecords(ctx context.Context, req *FinalDealRecordsRequest) (*FinalDealRecordsResponse, error) {
+	records, err := s.Module.FinalDealRecords()
+	if err != nil {
+		return nil, err
+	}
+	return &FinalDealRecordsResponse{Records: toRPCDealRecords(records)}, nil
+}
+
+// PendingDealRecords calls deals.PendingDealRecords.
+func (s *RPC) PendingDealRecords(ctx context.Context, req *PendingDealRecordsRequest) (*PendingDealRecordsResponse, error) {
+	records, err := s.Module.PendingDealRecords()
+	if err != nil {
+		return nil, err
+	}
+	return &PendingDealRecordsResponse{Records: toRPCDealRecords(records)}, nil
+}
+
+// AllDealRecords calls deals.AllDealRecords.
+func (s *RPC) AllDealRecords(ctx context.Context, req *AllDealRecordsRequest) (*AllDealRecordsResponse, error) {
+	records, err := s.Module.AllDealRecords()
+	if err != nil {
+		return nil, err
+	}
+	return &AllDealRecordsResponse{Records: toRPCDealRecords(records)}, nil
+}
+
+// RetrievalRecords calls deals.RetrievalRecords.
+func (s *RPC) RetrievalRecords(ctx context.Context, req *RetrievalRecordsRequest) (*RetrievalRecordsResponse, error) {
+	records, err := s.Module.RetrievalRecords()
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]*RetrievalRecord, len(records))
+	for i, r := range records {
+		ret[i] = &RetrievalRecord{
+			Addr: r.Addr,
+			Time: r.Time,
+			RetrievalInfo: &RetrievalInfo{
+				PieceCid:                r.RetrievalInfo.PieceCID.String(),
+				Size:                    r.RetrievalInfo.Size,
+				MinPrice:                r.RetrievalInfo.MinPrice,
+				PaymentInterval:         r.RetrievalInfo.PaymentInterval,
+				PaymentIntervalIncrease: r.RetrievalInfo.PaymentIntervalIncrease,
+				Miner:                   r.RetrievalInfo.Miner,
+				MinerPeerId:             r.RetrievalInfo.MinerPeerID,
+			},
+		}
+	}
+	return &RetrievalRecordsResponse{Records: ret}, nil
+}
+
+func toRPCDealRecords(records []deals.DealRecord) []*DealRecord {
+	ret := make([]*DealRecord, len(records))
+	for i, r := range records {
+		ret[i] = &DealRecord{
+			Addr:    r.Addr,
+			Time:    r.Time,
+			Pending: r.Pending,
+			DealInfo: &DealInfo{
+				ProposalCid:     r.DealInfo.ProposalCid.String(),
```

Comment:
Q: Should `r.DealInfo` be nil checked?

Reply:
I think it's fine since there are no pointers involved, just structs which will have default empty values.